lm_eval/base.py

```diff
@@ -186,7 +188,22 @@ def loglikelihood(self, requests):
 
     def loglikelihood_rolling(self, requests):
         # TODO: Implement caching once we've confirmed the perplexity implementation
-        # TODO: automatic batch size detection for vectorization
+        
+        # automatic batch size detection for vectorization
+        adaptive_batch_size = None
+        if self.batch_size == 'auto': 
+            # using rolling window with maximum context
+            print('Passed argument batch_size = auto. Detecting largest batch size')
+            @find_executable_batch_size(starting_batch_size=512) # if OOM, then halves batch_size and tries again
+            def forward_batch(batch_size):
+                test_batch = torch.ones((batch_size, self.max_length), device=self.device).long()
+                for _ in range(5): 
+                    out = F.log_softmax(self._model_call(test_batch), dim = -1).cpu()
+                return batch_size
+            
+            batch_size = forward_batch() 
+            print(f"Determined Largest batch size: {batch_size}")
+            adaptive_batch_size = batch_size
```

Comment:
I have tested this and it works correctly. You can also add this logic to the `greedy_until` method in `huggingface.py` and it works.

Reply:
I added this in. What tasks did you test this on? Just want to make sure everything works

Reply:
I just realized that you added the wrong code to the `greedy_until` method in `huggingface.py`. The correct code is the one in `loglikelihood_rolling` method.